tutorial.md

```diff
@@ -0,0 +1,202 @@
+``` 
+Hello, scylla-cdc!
+Hello, scylla-cdc!
+Hello, scylla-cdc!
+```
+The application printed one line for each CDC log consumed. 
+Now we can proceed and see how to use `CDCRow` struct to process the data.
+
+## Using CDCRow
+Having learned how to run the log reader, we can finally create the main logic of our application.
+We will edit the `consume_cdc` function of the `TutorialConsumer`.
+
+In general, there are four types of columns in CDC. You can access them in the following way:
+
+- CDC metadata columns can be accessed directly as member variables (e.g. `data.stream_id`)
+- values of columns can be accessed by method `get_value` or `take_value`
+- to check if a column was deleted in given operation, use `is_value_deleted`
+- to check which elements were deleted from a collection, use `get_deleted_elements` or `take_deleted_elements`
+
+The data returned is of type [CqlValue](<https://docs.rs/scylla/latest/scylla/frame/response/result/enum.CqlValue.html>)
+from the driver.
+
+For more detailed information about how to interpret these values, 
+check the [CDC documentation](<https://docs.scylladb.com/using-scylla/cdc/cdc-log-table/>).
+
+[//]: # (TODO: Add link to the API documentation)
+It is assumed that the user knows the metadata of their table, 
+but they can check if such a column exists in the CDC log, e.g. with method `column_exists`. 
+Refer to the API documentation to find an appropriate method.
+
+Back to our application, we will add printing of the data from the table:
+```rust
+async fn consume_cdc(&mut self, mut data: CDCRow<'_>) -> anyhow::Result<()> {
+    println!("_________");
+    println!("Time UUID: {:?}, operation type: {:?}", data.time, data.operation);
+    println!("pk: {}, ck: {}",
+             data.take_value("pk").unwrap().as_int().unwrap(),
+             data.take_value("ck").unwrap().as_int().unwrap());
+    let v = match data.take_value("v") {
+        Some(val) => Some(val.as_int().unwrap()),
+        None => None,
+    };
+    println!("v: {:?}, was deleted: {}",
+             v,
+             data.is_value_deleted("v"));
+    let vs = match data.take_value("vs") {
+        Some(val) => Some(val.into_vec().unwrap()),
+        None => None,
+    };
+    println!("vs: {:?}, was deleted: {}, deleted elements: {:?}",
+             vs,
+             data.is_value_deleted("vs"),
+             data.take_deleted_elements("vs"));
+    println!("_________");
+    Ok(())
+}
+```
+An example of the output:
+``` 
+_________
+Time UUID: 41183ab8-d07c-11ec-d982-2470743b0298, operation type: RowUpdate
+pk: 7, ck: 5
+v: None, was deleted: false
+vs: None, was deleted: false, deleted elements: [Int(6)]
+_________
+_________
+Time UUID: cc07e47a-d07c-11ec-ebc7-e1924abf3e16, operation type: RowUpdate
+pk: 1, ck: 2
+v: Some(8), was deleted: false
+vs: None, was deleted: false, deleted elements: []
+_________
+```
+
+<b>Note:</b> It is not guaranteed that the `CDCRow` struct will be alive after the function is terminated,
```

Comment:
I think it could be phrased better. It's not that "it's not guaranteed that the CDCRow will be alive" - the CDCRow's lifetime _does not allow it to exist_ after the function terminates.